tools/kicad/kicad_rs/src/eval/display.rs

```diff
@@ -0,0 +1,109 @@
+use evalexpr::Value;
+use std::fmt::{Display, Formatter, Result};
+
+const UNITS_LARGE: &[char] = &['k', 'M', 'G', 'T', 'E'];
+const UNITS_SMALL: &[char] = &['m', 'µ', 'n', 'p'];
+
+pub struct PrettyPrintValue<'a> {
+    value: &'a Value,
+    unit: &'a Option<String>,
+}
+
+impl<'a> PrettyPrintValue<'a> {
+    pub fn new(value: &'a Value, unit: &'a Option<String>) -> Self {
+        Self { value, unit }
+    }
+}
+
+impl<'a> Display for PrettyPrintValue<'a> {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        match self.value {
+            Value::Float(mut float) => {
+                let mut prefix = None;
+                if let Some((mut iter, mul)) = match float {
+                    _ if float == 0.0 => None,
+                    _ if float.abs() > 1000.0 => Some((UNITS_LARGE.iter(), 0.001)),
+                    _ if float.abs() < 1.0 => Some((UNITS_SMALL.iter(), 1000.0)),
+                    _ => None,
+                } {
+                    while let (i @ Some(_), true) =
+                        (iter.next(), float.abs() > 1000.0 || float.abs() < 1.0)
+                    {
+                        prefix = i;
+                        float *= mul;
+                    }
+                }
+
+                write!(f, "{:.2}", float)?; // Fixed two-decimal precision
```

Comment:
Fixing this to two decimal places is too precise for some components and carries can imply unreasonably high precision for the components. For example, generic 100 nF can have very large tolerances (around ± 20%) since their precise values are not critical to circuit performance. Having these specified as `100.00 nF` implies precision up to 0.01 (if only looking at the significant digits) which is way too precise considering that actual values from 80 nF to 120 nF would be within specification.

Reply:
I've now incorporated [`gpoint`](https://docs.rs/gpoint/0.1.3/gpoint/struct.GPoint.html) for proper significant digit formatting (that is lacking in Rust for now) and set it to use at most three significant digits. It does also nicely remove excess zeroes after the decimal point, for an example see the diff for [BD9E302EFJ_5V1.sch](https://github.com/racklet/electronics-prototyping/pull/27/commits/e6da60e90b3d0094b5b3770fac588fbbadf0d02d#diff-cb92efe1be69c8c7c78602cce855ad0a253369b168e26fdc078afd6cb5462d4b).